### PR TITLE
[python] Add unit tests for obsm, obsp, and to_anndata

### LINKED ISSUE #1930: [python] `obsm` and `varm` in `ExperimentQuery`
* https://github.com/single-cell-data/SOMA/issues/177 in the `somacore` repo
* Tag 1.0.5 of that repo
* Update this repo's `apis/python/setup.py` to depend on that

### PR DESCRIPTION
**Pull request by ebezzi**
**Issue and/or context:**
Resolves #1930 together with https://github.com/single-cell-data/SOMA/pull/179. Requires the latter to be released before tests can pass, so this is still marked as draft.

**Changes:**
This PR only contains the unit tests. All the code changes are in the SOMA repo.

**Notes for Reviewer:**

### CODE REVIEW

File: apis/python/tests/test_experiment_query.py [hunk 1]
```diff
@@ -547,6 +577,72 @@ def test_experiment_query_obsp_varp(soma_experiment):
             .concat()
         )
 
+        assert (
+            query.obsm("baz").tables().concat()
+            == soma_experiment.ms["RNA"]
+            .obsm["baz"]
+            .read((obs_slice, range(50)))
```
Comment (bkmartinjr) [inline apis/python/tests/test_experiment_query.py:584]: magic number 50 is in a few places.  Would be good to at least use a constant variable (e.g., N_FEAT or some such)

Comment (ebezzi) [reply apis/python/tests/test_experiment_query.py:584]: Good point, let me add it to a constant.

**Review by bkmartinjr — APPROVED** [review]
one minor nit in a comment - otherwise LGTM

**Review by thetorpedodog — APPROVED** [review]
Looks great! One thing I would recommend to make dev/review a bit easier is to substitute in the development revision into the setup.py file for the duration of the review, so it can be more easily evaluated during the review process, but once the upstream is released, you can cut it over to the final version and fully integrate everything.

**Pull request merged**